pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py

```diff
@@ -46,6 +53,135 @@ def test_sync_publish_update_info(self):
+
+        This test provides the following
+
+        1. Check whether all the modules in the published repo
+           contains a collection field
+        2. Check whether the collection field is unique
+        """
+        repo, update_list = self._set_repo_and_get_repo_data()
+
+        # getting the update info from the fixtures repo
+        update_info_fixtures = get_xml_content_from_fixture(
+            fixture_path=RPM_WITH_MODULES_FEED_URL,
+            data_type='updateinfo',
+        )
+
+        # collections from published repo
+        collection_update_list = {
+            update.find('./id').text: update.find('.//collection').attrib['short']
+            for update in update_list.findall('update')
+        }
+
+        # collections from fixtures
+        # the below dict comprehension computes the following
+        # It gets the collection elements from the fixtures `updateinfo`,
+        # gets the module names from each collection and maps it to update ID.
+        # If module name doesn't exist, it replaces it as default
+        # else module name is stored.
+        # example : collections_from_fixtures = {
+        #                                        'RHEA..1' : 'duck',
+        #                                        'RHEA..2' : 'default'
+        #                                       }
+        collections_from_fixtures = {
+            update.find('id').text:
+            'default' if update.find('.//module') is None
+            else update.find('.//module').attrib['name']
+            for update in
+            update_info_fixtures.findall('.//update')
+        }
+        # Adding repo_id, indexes and module_name for the collection
+        # example : collections_from_fixtures = {
+        #                                        'RHEA..1' : 'repo_id_1_duck',
+        #                                        'RHEA..2' : 'repo_id_2_duck'
+        #                                       }
+        indexes = {}
+        for key in collections_from_fixtures:
+            val = collections_from_fixtures[key]
+            if val in indexes:
+                indexes[val] += 1
+            else:
+                indexes[val] = 1
+            collections_from_fixtures[key] = '{}_0_default'.format(repo['id']) \
+                if val == 'default' \
+                else '{}_{}_{}'.format(repo['id'], indexes[val], val)
+
+        self.assertEqual(
+            collections_from_fixtures,
+            collection_update_list,
+            'collection names not proper'
+        )
+
+    def test_copy_errata(self):
+        """Test whether Errata modules are copied.
+
+        This Test does the following.
+
+        1. It creates,syncs, and publishes a modules rpm repository
+        2. Creates another repo with no feed
+        3. Recursively copies an errata from one repo to another
+        4. Checks whether the errata information in the new repo is
+        correct
```

Comment:
`rst` can be very pick. I suggest to take a look into the generated HTML to verify if the generated docs are as expected. Run `make docs-html`

This site can be helpful http://rst.ninjs.org/

```.rst
        1. It creates,syncs, and publishes a modules rpm repository
        2. Creates another repo with no feed
        3. Recursively copies an errata from one repo to another
        4. Checks whether the errata information in the new repo is
           correct
```